logical/plugin/plugin.go

```diff
@@ -0,0 +1,97 @@
+package plugin
+
+import (
+	"fmt"
+
+	"sync"
+
+	"github.com/hashicorp/go-plugin"
+	"github.com/hashicorp/vault/helper/pluginutil"
+	"github.com/hashicorp/vault/logical"
+)
+
+// PluginMap should be used by clients for map of plugins
+var PluginMap = map[string]plugin.Plugin{
```

Comment:
It's not clear what this is here for. I see where it's used but it essentially seems to be holding a single implementation that will never change and never be added to? Since anything else would derive from BackendPlugin.